behavior/datagen/gen.py

```diff
@@ -540,21 +541,35 @@ def main(self):
             # Copy the config file to the output directory.
             shutil.copy(config_path, mode_dir)
 
+            # build sweeping space.
+            ps_space = param_sweep_space(self.config['param_sweep'])
             # For each benchmark, ...
             for benchmark in benchmarks:
-                # Run BenchBase to generate training data.
-                results_dir = Path(mode_dir / benchmark)
-                results_dir.mkdir()
-                benchbase_results_dir = Path(results_dir / "benchbase")
-                benchbase_results_dir.mkdir(exist_ok=True)
-                logger.info(
-                    "Running experiment %s with benchmark %s and results %s",
-                    experiment_name,
-                    benchmark,
-                    results_dir,
-                )
-                self.run_experiment(benchmark, results_dir, benchbase_results_dir)
-
+                input_cfg_path = self.dir_benchbase_config / f"{benchmark}_config.xml"
+
+                # func def for recursive sweeping.
+                def sweep_func(params):
+                    # assemble the result dir given params
```

Comment:
```suggestion
                    # Assemble the result directory from the parameters.
```

General note: documentation and line spacing is cheap, so I prefer to avoid shortening names "dir", "params", unless there is a good reason for doing so.